### PR TITLE
Feature/sflow input module

### LINKED ISSUE #106: sFlow input module
https://sflow.org/developers/specifications.php

reference https://github.com/bio-routing/flowhouse

https://github.com/sflow/host-sflow

https://github.com/sflow/sflowtool

https://github.com/cloudflare/goflow/tree/master/decoders

https://github.com/pavel-odintsov/fastnetmon/tree/master/src

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/handlers/net/NetStreamHandler.cpp [hunk 1]
```diff
@@ -403,6 +416,106 @@ void NetworkMetricsBucket::process_dnstap(bool deep, const dnstap::Dnstap &paylo
...
+        case IP_PROTOCOL::UDP:
+            ++_counters.UDP;
+            break;
+        default:
+            ++_counters.OtherL4;
+            break;
+        }
+
+        if (sample.ifCounters.ifDirection == DIRECTION::IN) {
+            ++_counters.total_in;
```
Comment (weyrick) [inline src/handlers/net/NetStreamHandler.cpp:443]: i believe we'd need to not just `++` but `+=` the amount of packets that the sample saw - it's almost definitely not just one?

File: src/inputs/sflow/README.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Sflow Stream Input
+
+This directory contains a sflow/example stream input tap, suitable only for testing.
```
Comment (weyrick) [inline src/inputs/sflow/README.md]: `This directory contains the sflow input tap`

File: src/inputs/sflow/SflowInputStream.cpp [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+        sample.rawSampleLen = event.length;
+        sample.sourceIP.type = SFLADDRESSTYPE_IP_V4;
+        struct sockaddr_in peer4;
+        inet_pton(AF_INET, event.sender.ip.c_str(), &(peer4.sin_addr));
+        std::memcpy(&sample.sourceIP.address.ip_v4.addr, &peer4.sin_addr, 4);
+        try {
+            read_sflow_datagram(&sample);
+            sflow_signal(sample);
+        } catch (const std::exception &e) {
+            ++_error_count;
```
Comment (weyrick) [inline src/inputs/sflow/SflowInputStream.cpp:122]: do we expose this anywhere?

File: src/inputs/sflow/SflowInputStream.cpp [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+            _io_thread->join();
+        }
+    }
+
+    _running = false;
+}
+
+void SflowInputStream::info_json(json &j) const
+{
+    common_info_json(j);
```
Comment (weyrick) [inline src/inputs/sflow/SflowInputStream.cpp:155]: we can expose error_count here if nowhere else.

**Review by weyrick — CHANGES_REQUESTED** [review]
also:
- README
- docker/* changes
- GitHub CI changes

File: README.md [hunk 5]
```diff
@@ -59,7 +58,7 @@ and [DNS](src/handlers/dns) stream processors, and the types of summary informat
 
 One of the easiest ways to get started with pktvisor is to use
 the [public docker image](https://hub.docker.com/r/ns1labs/pktvisor). The image contains the collector
-agent (`pktvisord`), the command line UI (`pktvisor-cli`), the pcap file analyzer (`pktvisor-pcap`), and the dnstap file analyzer (`pktvisor-dnstap`). When running
+agent (`pktvisord`), the command line UI (`pktvisor-cli`), the pcap and dnstap file analyzer (`pktvisor-reader`), and the dnstap file analyzer (`pktvisor-dnstap`). When running
```
Comment (weyrick) [inline README.md]: need to cut the second "and the dnstap file analyzer (`pktvisor-dnstap`)" part

**Review by weyrick — APPROVED** [review]

File: src/handlers/net/NetStreamHandler.cpp (hunk 1 above)
Comment (ghost) [reply src/handlers/net/NetStreamHandler.cpp:443]: ifDirection is a enum

File: src/inputs/sflow/SflowInputStream.cpp [hunk 6]
```diff
@@ -153,6 +153,7 @@ void SflowInputStream::stop()
 void SflowInputStream::info_json(json &j) const
 {
     common_info_json(j);
+    j["input"]["packet_error"] = _error_count.load();
```
Comment (weyrick) [inline src/inputs/sflow/SflowInputStream.cpp]: see https://github.com/ns1labs/pktvisor/blob/b46517133d82b3ac4d8cbdfff0d78618cbeb68c7/src/inputs/pcap/PcapInputStream.cpp#L581

this needs to be prefixed with schema_key(). in that case you can leave out "input" and just make it `j[schema_key()]["packet_errors"]` (note plural)

**Review by weyrick — CHANGES_REQUESTED** [review]

**Review by weyrick — APPROVED** [review]

**Pull request merged**